### ISSUE TITLE
[LOTTERY] As long as no participants are delinquent, remove startRound time constraint

### ISSUE THREAD

**Issue by blorange** [WeTrustPlatform/rosca-contracts#166]
This was just a general thought, but it would remove any concerns about wasting gas, and means there would be less 'checking in' to see when it was possible to start the next round. It wouldn't be very fair in a bidding Circle, but in a lottery or a predefined one I think this feature would be useful.

WDYT @shine2lay @rmerom?

**Comment by rmerom** [comment]
I think we can work more on the whole round thing. What do you think is the idea round-advancement mechanism? I'd think it should be "automatic", so that ROSCAs are deterministic and it's easy for each participant to understand when a round starts and when it ends. What do you think?

**Comment by blorange** [comment]
Yeah I agree - I do think automatic is better. The only concern is the massively variable gas costs if we want to bake `startRound` into the existing functions.

What about having the round-processing functions moved over to the 'withdraw' function? So the amounts for the user are only calculated on withdrawal? It might not save on gas, but it would mean that we're trying to have constant gas costs for each function, which IMO is somewhat important.
